src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -249,7 +255,15 @@ Interop.NetSecurityNative.Status status
                     throw new Interop.NetSecurityNative.GssApiException(status, minorStatus);
                 }
 
-                return Encoding.UTF8.GetString(token.ToByteArray());
+                byte[] tokenBytes = token.ToByteArray();
+                int length = tokenBytes.Length;
+                if (length > 0 && tokenBytes[length - 1] == '\0')
+                {
+                    // Some GSS-API providers (gss-ntlmssp) include the terminating null with strings, so skip that.
+                    --length;
+                }
+
+                return Encoding.UTF8.GetString(tokenBytes, 0, length);
```

Comment:
Consider adding a `Span` property to `GssBuffer` to avoid the throwaway copy (`ToByteArray`) here.

Reply:
I'll try adding a method/property to return a `ReadOnlySpan`. But `GssBuffer` currently uses an internal `IntPtr`. And based on [this](https://github.com/dotnet/corefx/issues/14507#issuecomment-267235430), there is no constructor to convert `IntPtr` to Span<T> type. But the suggestion is to cast it (void*).

Reply:
I think `ReadOnlySpan<byte> Span => _data != IntPtr.Zero && _length != 0 ? new ReadOnlySpan<byte>(_data.ToPointer(), checked((int)_length)) : default;` would be sufficient.

Reply:
Thanks. That will work.